build/gutenberg-block.js

```diff
@@ -1 +1,154 @@
-(()=>{"use strict";const e=window.React,{registerBlockType:t}=wp.blocks,{SelectControl:l}=wp.components,{useState:n,useEffect:s}=wp.element;t("business-profile-render/bpr-block",{title:"Business Profile",icon:"admin-site",category:"common",attributes:{option:{type:"string",default:"--"}},edit:t=>{const{attributes:a,setAttributes:o}=t,{option:i}=a,[r,c]=n([]);return s((()=>{const e=window.businessProfileData||{},t=Object.entries(e).map((([e,t])=>({label:e,value:t})));c(t)}),[]),(0,e.createElement)("div",null,(0,e.createElement)(l,{label:"Select which field you want to show",value:i,options:r.map((e=>({label:e.label,value:e.value}))),onChange:e=>{o({option:e})}}),(0,e.createElement)("p",null,(0,e.createElement)("span",{class:"css-1imalal"},"Preview:"),(0,e.createElement)("br",null),i))},save:({attributes:t})=>{const{option:l}=t;return(0,e.createElement)("div",null,(0,e.createElement)("p",null,l))}})})();
\ No newline at end of file
+(() => {
+    "use strict";
+    const e = window.React,
+        { registerBlockType: t } = wp.blocks,
+        { SelectControl: l } = wp.components,
+        { useState: n, useEffect: s } = wp.element;
+
+    t("business-profile-render/bpr-block", {
+        title: "Business Profile",
+        icon: "admin-site",
+        category: "common",
+        attributes: {
+            option: { type: "string", default: "--" }
+        },
+        edit: t => {
+            const { attributes: a, setAttributes: o } = t,
+                { option: i } = a,
+                [r, c] = n([]);
+
+            s(() => {
+                const e = window.businessProfileData || {},
+                    t = Object.entries(e).map(([e, t]) => ({ label: e, value: t }));
+                c(t);
+            }, []);
+
+            const isImageUrl = (url) => {
+                try {
+                    // Remove trailing commas before parsing the URL
+                    const cleanUrl = url.replace(/,+$/, '');
+                    const parsedUrl = new URL(cleanUrl);
+                    return parsedUrl.hostname === 'media-prod.apigateway.co';
+                } catch (e) {
+                    return false;
+                }
+            };
+
+            const getSocialIcon = (url) => {
+                try {
+                    const parsedUrl = new URL(url);
+                    if (parsedUrl.hostname.includes("facebook.com")) {
```

Comment:
Can you use a switch case here?

Reply:
@Padminia Replaced the try-catch block with a switch-case statement for handling social links.